src/imap/scan_folders.rs

```diff
@@ -85,6 +86,34 @@ impl Imap {
         last_scan.replace(Instant::now());
         Ok(())
     }
+
+    pub async fn list_folders_except(
+        self: &mut Imap,
+        context: &Context,
+        exclude: &[impl AsRef<str>],
+    ) -> Result<Vec<String>> {
+        let exclude = &exclude.iter().map(|s| s.as_ref()).collect::<Vec<_>>();
+        let session = self.session.as_mut();
+        let session = session.context("list_folders_except() IMAP No Connection established")?;
+        let list = session
+            .list(Some(""), Some("*"))
+            .await?
+            .filter_map(|f| async move {
+                if let Err(e) = &f {
+                    warn!(context, "list_folders_except can't get folder: {}", e);
+                };
+                f.ok()
+            })
+            .filter_map(|f| async move {
+                let name = f.name();
+                if exclude.contains(&name) {
```

Comment:
Could probably use `exclude.iter().any(|s| s.as_ref() == name)` here over original `exclude` without allocating a new `Vec` for `as_ref()` results at the top of the function.